### PR TITLE
Add pre-commit hook that runs tests and prettier

### PR DESCRIPTION
**Pull request by MattRC7**
Installs `husky` and `prettier` and runs tests + validates formatting before commits. Adds `husky` to the `postinstall` npm script so hooks will be added whenever `npm ci` or `npm install` is run.

Also fixes any prettier formatting issues.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
-          command: npm run build
-      - run:
-          name: npm test
-          command: npm test
-      - save_cache:
-          paths:
-            - ~/.npm
-            - node_modules
-          key: livgust-macovidvaccines.com-{{ checksum "package.json" }}2
+    build:
```
Comment (j256) [inline .circleci/config.yml:4]: Why make all of these indenting changes?  It obscures the real diffs here.

Comment (MattRC7) [reply .circleci/config.yml:4]: Without it, the prettier pre-commit hook fails. Part of the intent of this PR is to ensure prettier has been run before committing, so I had to run it and make sure all files comply.

If you think a file should be excluded from formatting let me know, and i can add it to the .prettierignore file.

**Comment by livgust** [comment]
I'm getting errors from `prettier` for:
```
Checking formatting...
[warn] .circleci\config.yml
[warn] .prettierrc.json
[warn] build\asset-manifest.json
[warn] build\index.html
[warn] build\manifest.json
[warn] build\static\css\main.7dc8eeae.chunk.css
[warn] build\static\js\2.253850ed.chunk.js
[warn] build\static\js\3.e8bff064.chunk.js
[warn] build\static\js\main.55354423.chunk.js
[warn] build\static\js\runtime-main.ce8974df.js
[warn] CONTRIBUTING.md
[warn] package-lock.json
[warn] package.json
[warn] README.md
[warn] src\App.js
[warn] src\components\Availability.js
[warn] src\components\Availability.test.js
[warn] src\components\HelpDialog.js
[warn] src\components\Menu.js
[warn] src\components\MoreInformation.js
[warn] src\components\MoreInformation.test.js
[warn] src\components\SignUpLink.js
[warn] src\components\StaleDataIndicator.js
[warn] src\components\StaleDataIndicator.test.js
[warn] src\components\StateEligibility.js
[warn] src\CovidAppointmentTable.js
[warn] src\CovidAppointmentTable.test.js
[warn] src\index.css
[warn] src\index.js
[warn] src\reportWebVitals.js
[warn] src\setupTests.js
[warn] src\theme.js
[warn] test\fixtures\api\no-data.json
[warn] test\fixtures\api\prod.json
[warn] Code style issues found in the above file(s). Forgot to run Prettier?
```

1) we should add `build` to the ignore file
2) I ran prettier on index.js and nothing changed, so I don't know why I'm getting errors for that file (and I'm guessing the rest are about the same)

`eslint` worked fine for me :)

**Review by livgust — CHANGES_REQUESTED** [review]
See above comment

**Comment by MattRC7** [comment]
Ooh that's weird, i'll look into that. Will definitely add build to the ignore list, also package_lock.json and any other generated files.

Frankly, i can limit it to things inside `src` if that makes more sense.

**Comment by livgust** [comment]
Thanks! Everything-besides-src sounds good but go with whatever you think is best! And LMK if you need to screen share - any time after 8 lol

**Comment by MattRC7** [comment]
@livgust okay just pushed updates! And then I ran this and all looks good on my end: ```
npm run check:prettier

> ma_covid_vaccine_website@0.1.0 check:prettier /home/matt/projects/macovidvaccines.com
> prettier --check ./src

Checking formatting...
All matched files use Prettier code style!
```

If you're still seeing failures it must be due to environment differences and we can hop on screenshare to debug.

**Review by livgust — APPROVED** [review]

**Pull request merged**